src/main/java/org/tron/common/storage/DepositImpl.java

```diff
@@ -302,7 +302,14 @@ public synchronized Storage getStorage(byte[] address) {
 
     Storage storage;
     if (this.parent != null) {
-      storage = parent.getStorage(address);
+      Storage parentStorage = parent.getStorage(address);
+      if (Args.getInstance().getBlockVersion() == 3) {
+        // old logic
+        storage = parentStorage;
+      } else {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move this "else" keyword to a new dedicated line. ([squid:RightCurlyBraceDifferentLineAsNextBlockCheck](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3ARightCurlyBraceDifferentLineAsNextBlockCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1664&issues=AWbJU6pnSgN4uCclo1cW&open=AWbJU6pnSgN4uCclo1cW)

[ISSUE_KEY]: AWbJU6pnSgN4uCclo1cW